tests/jax/components/executing/observing_test.py

```diff
@@ -0,0 +1,272 @@
+    """Mock executor component."""
+    return MockExecutor()
+
+
+@pytest.fixture
+def mock_feedforward_executor_observe() -> FeedforwardExecutorObserve:
+    """Mock FeedforwardExecutorObserve.
+
+    Returns:
+        FeedforwardExecutorObserve
+    """
+    return FeedforwardExecutorObserve()
+
+
+# Test on_execution_observe_first
+def test_on_execution_observe_first_without_adder(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor_without_adder: Executor,
+) -> None:
+    """Test entering executor without store.adder
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor_without_adder: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe_first(
+        executor=mock_executor_without_adder
+    )
+
+    assert not mock_executor_without_adder.store.adder
+
+
+def test_on_execution_observe_first(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor: MockExecutor,
+) -> None:
+    """Test on_execution_observe_first method from FeedForwardExecutorObserve
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe_first(executor=mock_executor)
+
+    assert not mock_executor.store.network_int_keys_extras == None
+    assert (
+        mock_executor.store.extras["network_int_keys"]
+        == mock_executor.store.network_int_keys_extras
+    )
+    assert mock_executor.store.adder.parm == "after_add_first"
+
+
+# test on_execution_observe
+def test_on_execution_observe_without_adder(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor_without_adder: Executor,
+) -> None:
+    """Test entering executor without store.adder
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor_without_adder: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe(
+        executor=mock_executor_without_adder
+    )
+
+    assert not mock_executor_without_adder.store.adder
+
+
+def test_on_execution_observe(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor: MockExecutor,
+) -> None:
+    """Test on_execution_observe method from FeedForwardExecutorObserve
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe(executor=mock_executor)
+
+    for agent in mock_executor.store.policies_info.keys():
+        assert mock_executor.store.next_extras["policy_info"][
+            agent
+        ] == "policy_info_" + str(agent)
+    assert (
+        mock_executor.store.next_extras["network_int_keys"]
+        == mock_executor.store.network_int_keys_extras
+    )
+    assert mock_executor.store.adder.parm == "after_add"
```

Comment:
It would be nice to change the `mock_adder` methods so that you can see that the correct arguments are being passed through to the adder, instead of just checking that the method was called.

Reply:
I agree with this idea.